src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -460,14 +460,7 @@ public static uint CountSignificantBits(uint value)
 
             public static uint CountSignificantBits(ulong value)
             {
-                uint upper = (uint)(value >> 32);
-
-                if (upper != 0)
-                {
-                    return 32 + CountSignificantBits(upper);
-                }
-
-                return CountSignificantBits((uint)value);
+                return 64 - (uint)BitOps.LeadingZeroCount(value);
```

Comment:
Missed optimization from previous PR

Reply:
Could you please submit this as separate PR?

Reply:
Done - https://github.com/dotnet/coreclr/pull/22612